src/solvers/sat/satcheck_minisat2.cpp

```diff
@@ -328,14 +328,16 @@ bool satcheck_minisat2_baset<T>::is_in_conflict(literalt a) const
 template<typename T>
 void satcheck_minisat2_baset<T>::set_assumptions(const bvt &bv)
 {
-  assumptions=bv;
-
-  forall_literals(it, assumptions)
-    if(it->is_true())
+  // We filter out 'true' assumptions which cause an assertion violation
+  // in Minisat2.
+  assumptions.clear();
+  forall_literals(it, bv)
```

Comment:
If you're refactoring this, would you mind using a ranged for instead of this macro?